src/ngx_http_lua_module.c

```diff
@@ -746,6 +746,12 @@ ngx_http_lua_init(ngx_conf_t *cf)
     if (lmcf->lua == NULL) {
         dd("initializing lua vm");
 
+        /* firstly set environment for lua vm */
+        if (ngx_set_environment(cf->cycle, NULL) == NULL) {
```

Comment:
I'm worried about this approach since the system env would needs to roll back to its old value when the config loading process turns out to fail in the end (HUP reload may fail) and we may end up with a modified master which may fork off bad nginx worker processes using the old configuration.

Maybe it's safer to override `os.getenv` with our own FFI-based Lua API function implementation?